### ISSUE TITLE
Make a test that the UTXO set tree is not modified as a result of proofsForTransactions call

### ISSUE THREAD

**Issue by kushti** [ergoplatform/ergo#1614]
While the current 
```
property("concurrent applyModifier() and proofsForTransactions()")
```

test ensures that ```proofsForTransactions()``` and ```applyModifier()``` can work in parallel , it is not checking that tree is not modified in some edge cases. Actually, it could be the case that it happens in some edge cases.

Make a test which is generating e.g. 500 blocks of 500 txs each as testing samples. Call proofsForTransactions first, check that digest from its output corresponds to header's stateRoot, and header.ADProofsRoot corresponds to ```ADProofs.proofDigest(proofBytes)```. Then call applyModifier and check it is finishing successfully

**Comment by dimension-drifter** [comment]
@kushti i have raised a PR for this issue. Please Review it

**Comment by Ergologica** [comment]
Is the gap here the scale rather than the invariant itself? On current master `UtxoStateSpecification` already carries two properties that assert it:

- `proofsForTransactions() does not change state digest` (line 471) - takes the root digest, calls `proofsForTransactions` twice, and asserts the digest is unchanged after each call;
- `proofsForTransactions() to be deterministic` (line 289) - same transactions in, same proof digest and state digest out.

Both run against a state freshly built from a `boxesHolderGen` sample. What they do not cover is the shape this issue sizes it at: the invariant holding on a tree that has already been mutated many times by `applyModifier`, rather than on a newly created one - which is where an in-place mutation would actually show up.

I have that version written and green: root digest and version checked around every `proofsForTransactions` call along a chain, with repeated and interleaved calls at each height, and a check that applying the block moves the root exactly to the digest the proof announced. Happy to open it as a PR if that is the missing piece.

Asking first rather than sending a PR because this is still listed as an open 200 SigUSD bounty in the ErgoDevs registry, and if you consider the invariant already covered by the two properties above then the issue should be closed instead - I would rather not claim a bounty that is already satisfied.
